proposals/web-api-add-edit-with-media.md

```diff
@@ -0,0 +1,618 @@
+* Image failed copyright ID check
+* Content of the media does not meet compliance rules
+
+## Media Add/Edit Process
+
+The creation of the media resource will be done using as a standard RESO Web API 2.1.0 Add/Edit process. The post SHOULD include all known resource record fields for the media with the exception of only the media byte stream.
+
+The response object MAY contain `mediaReadLink` and `mediaEditLink` annotations which the client MUST respect as per the OData specification when communicating the byte stream to the server. If fields are not provided the implicit URLs per the OData specification MUST then be used.  
+
+Byte stream endpoints MUST only expect previously established security credentials if they are in the same HTTP security domain as the original request.  Byte stream endpoints outside the original OData security domain cannot expect any security credentials to be provided outside of the contents of the respective annotation being used.
+
+The Byte stream endpoint will provide HTTP response code values related to the success of the byte stream POST.  All HTTP 2xx response codes are successful.  All HTTP 4xx error codes except for HTTP 409 (Conflict) SHOULD have the client query the OData resource to validate/refresh the annotation URL and retry the POST using the URL.
+
+The HTTP 409 (Confict) response designates that the byte stream cannot be replaced.  This can happen if the implementation is using a write-once model for media to retain history.  When a 409 is received, the client SHOULD add a new media record with the updated byte stream and then delete the original Media record.
+
+## Media Post Processing
+
+The Media record will transition to `Processing` status while the server prepares the media for distribution downstream. The implementation will do all the work required for distribution before changing the status to `Complete`
+
+If any post-upoload processing fails, the media record will be have a status of `Rejected` and the reasoning for the Media submitter will be populated in the `MediaStatusDescription` field for actions to be taken.
+
+Rules MAY be written against the `MediaStatus` field to handle the order of operations required to publish a listing.  Example:  Media must be complete before it can be attached to a listing or Media attached to a listing must all be Complete before a listing can be on-market.
+
+## Media Add/Edit Examples
+
+### Create Initial Media Resource
+Create the initial media record using the OData standard. This could be an expanded record on another resource as a secondary approach but the example will be with Media as a tier 1 Resource/Model.
+
+**REQUEST**
+```http
+POST https://api.my-webapi.io/Media
+OData-Version: 4.01
+Content-Type: application/json
+Accept: application/json
+Prefer: return=representation
+```
+```json
+{
+ "Caption": "Ipsum Lorum",
+ "Order": 1,
+ "ImageType": "image/jpeg",
+ 
+}
+```
+
+**RESPONSE - External Media storage**
```

Comment:
Should break up these two cases into two sections based on the response rather than just two responses.